File: database/database-annotation-processor/src/test/java/ru/tinkoff/kora/database/common/annotation/processor/vertx/VertxExtensionTest.java
package ru.tinkoff.kora.database.common.annotation.processor.vertx;

import org.junit.jupiter.api.Test;
import ru.tinkoff.kora.annotation.processor.common.TestUtils;
import ru.tinkoff.kora.application.graph.TypeRef;
import ru.tinkoff.kora.database.common.annotation.processor.entity.TestEntityJavaBean;
import ru.tinkoff.kora.database.common.annotation.processor.entity.TestEntityRecord;
import ru.tinkoff.kora.database.common.annotation.processor.vertx.VertxEntity.AllNativeTypesEntity;
import ru.tinkoff.kora.database.vertx.mapper.result.VertxResultColumnMapper;
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowMapper;
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowSetMapper;

import java.util.List;

public class VertxExtensionTest {
    @Test
    void testEntity() throws Exception {
        TestUtils.testKoraExtension(new TypeRef<?>[]{
                TypeRef.of(VertxRowSetMapper.class, TypeRef.of(List.class, TestEntityRecord.class)),
                TypeRef.of(VertxRowMapper.class, TestEntityRecord.class),
                TypeRef.of(VertxRowSetMapper.class, TypeRef.of(List.class, TestEntityJavaBean.class)),
                TypeRef.of(VertxRowMapper.class, TestEntityJavaBean.class),
                TypeRef.of(VertxRowSetMapper.class, TypeRef.of(List.class, AllNativeTypesEntity.class)),
                TypeRef.of(VertxRowMapper.class, AllNativeTypesEntity.class),
            },
            TypeRef.of(VertxResultColumnMapper.class, TestEntityRecord.UnknownTypeField.class),
            TypeRef.of(VertxEntity.TestEntityFieldVertxResultColumnMapperNonFinal.class)
        );
    }
}

File: database/database-symbol-processor/src/test/kotlin/ru/tinkoff/kora/database/symbol/processor/vertx/VertxExtensionTest.kt
package ru.tinkoff.kora.database.symbol.processor.vertx

import org.junit.jupiter.api.Test
import ru.tinkoff.kora.database.symbol.processor.entity.EntityWithEmbedded
import ru.tinkoff.kora.database.symbol.processor.entity.TestEntity
import ru.tinkoff.kora.database.vertx.mapper.result.VertxResultColumnMapper
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowMapper
import ru.tinkoff.kora.database.vertx.mapper.result.VertxRowSetMapper
import ru.tinkoff.kora.ksp.common.TestUtils
import kotlin.reflect.typeOf

class VertxExtensionTest {
    @Test
    fun testEntity() {
        TestUtils.testKoraExtension(
            arrayOf(
                typeOf<VertxRowMapper<TestEntity>>(),
                typeOf<VertxRowMapper<AllNativeTypesEntity>>(),
                typeOf<VertxRowSetMapper<List<TestEntity>>>(),
                typeOf<VertxRowSetMapper<List<AllNativeTypesEntity>>>(),
                typeOf<VertxRowSetMapper<TestEntity>>(),
                typeOf<VertxRowSetMapper<AllNativeTypesEntity>>(),
                typeOf<VertxRowMapper<EntityWithEmbedded>>(),
                typeOf<VertxRowSetMapper<List<EntityWithEmbedded>>>(),
                typeOf<VertxRowSetMapper<EntityWithEmbedded>>(),
                typeOf<VertxRowSetMapper<List<String>>>(),
            ),
            typeOf<VertxResultColumnMapper<TestEntity.UnknownField>>(),
            typeOf<TestEntityFieldVertxResultColumnMapperNonFinal>(),
            typeOf<VertxRowMapper<String>>(),
        )
    }
}
